src/north/amazon/AmazonS3.class.spec.js

```diff
@@ -0,0 +1,54 @@
+const fs = require('fs')
+const path = require('path')
+const ApiHandler = require('../ApiHandler.class')
+const AmazonS3 = require('./AmazonS3.class')
+// const config = require('../../config/defaultConfig.json')
+const config = require('../../../tests/testConfig').default
+
+
+// Mock database service 
+jest.mock('../../services/database.service',() => {})
+
+// Mock logger
+jest.mock('../../engine/Logger.class')
+
+// Mock engine
+const engine = jest.genMockFromModule('../../engine/Engine.class')
+engine.configService = { getConfig: () => ({ engineConfig: config.engine }) }
+
+beforeEach(() => {
+jest.resetAllMocks()
+})
+
+describe('Amazone S3 north', () => {
+
+  it('should be properly initialized', () => {
+    const AmazonS3Config = config.north.applications[8]
+    const AmazonS3North = new AmazonS3(AmazonS3Config, engine)
+    expect(AmazonS3North.canHandleFiles).toBeTruthy()
+    // expect() tobeContinued
+  })
+
+
+  it('should properly handle file', async () => {
+    const AmazonS3Config = config.north.applications[8]
+    const AmazonS3North = new AmazonS3(AmazonS3Config, engine)
+    const filePath = '/path/to/file/example.file'
+    jest.spyOn(fs, 'createReadStream').mockImplementation(() => ({ key: '' }))
+
+    await AmazonS3North.handleFile(filePath)
+
+    // let expectedResult = null
+    let expectedError = null
+
+    try {
+    // expectedResult = await ...
+    } catch (error) {
+      expectedError = error
+    }
+
+    expect(AmazonS3North.params).toBeCalledWith(filePath, bucket, body, key)
```

Comment:
AmazonS3North.params is not a function so you cannot check that it is called with these parameters.

You can instead check for its values. Check on other component for this kind of test

Reply:
You have something similar juste two lines beneath (l 52)